File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/ChangeGameStatePacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record ChangeGameStatePacket(short reason, float value) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeByte(reason);
        os.writeFloat(value);
    }

    @Override
    public byte getId() {
        return 0x2B;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/ChangeGameStatePacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class ChangeGameStatePacket(val reason: Short, val value: Float) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeByte(reason.toInt())
        os.writeFloat(value)
    }

    override val id: Byte
        get() = 0x2B
}
